src/views/MainScreen.tsx

```diff
@@ -27,39 +31,77 @@ const styles = StyleSheet.create({
   }
 });
 
-export default function MainScreen() {
-  const userId = 1;
-  const [exerciseLabel, exerciseLabelUpdate] = useState('My exercise');
-  const [exerciseCalories, exerciseCaloriesUpdate] = useState('100');
-
+const getTomorrowString = (): string => {
   const tomorrow = new Date();
   tomorrow.setDate(tomorrow.getDate() + 1);
-  const { data: exercises, refetch: refetchExercises } = useGetExerciseRecords({ userId, queryParams: { endDateTime: tomorrow.toISOString(), startDateTime: '0000-00-00 00:00:00' } });
-  const { mutate: postExercise } = usePostExerciseRecord({ userId });
-
-  const submitExercise = async () => {
-    const timestamp = new Date();
-
-    const exercise: ExerciseRecord = {
-      calories: parseInt(exerciseCalories, 10),
-      label: exerciseLabel,
-      timestamp: timestamp.toISOString()
-    };
-
-    await postExercise(exercise);
-    exerciseLabelUpdate('My exercise');
-    exerciseCaloriesUpdate('100');
-    refetchExercises();
-  };
+  tomorrow.setUTCHours(0);
+  tomorrow.setUTCMinutes(0);
+  tomorrow.setUTCSeconds(0);
+  tomorrow.setUTCMilliseconds(0);
+
+  // Get the ISO string and massage it into the format expected by the server (remove T separator between date and
+  // time, and remove fractional milliseconds and the timezone indicator)
+  return tomorrow.toISOString().replace('T', ' ').slice(0, -5);
+};
+
 
+export default function MainScreen() {
+  const [exerciseLabel, setExerciseLabel] = useState('');
+  const [exerciseCalories, setExerciseCalories] = useState('0');
+  const startTime = '0000-00-00 00:00:00';
+
+  // Get tomorrow's date. Only recompute if today's date changes
```

Comment:
This seems slightly over-engineered for the benefit of switching over to a new set of data automatically at midnight (UTC time). The cost is that we are creating a new Date object and getting the UTC date from it with every render loop.

I recommend simplifying this for now and just calculating the start and end time for "today" outside of the component function. I imagine we'll build out UX allowing users to view data for different days later.

Reply:
I agree that this will probably change later, but I'd rather err on the side of correctness rather than optimizing performance now. I did end up simplifying the get tomorrow function itself, though - we no longer need to zero out the time because it's memoized and based on the date, so it won't re-run unless the date changes.